src/models/data.py

```diff
@@ -231,13 +321,15 @@ def __next__(self) -> Tuple[Tuple[Union[np.ndarray, torch.Tensor],
                                 Union[np.ndarray, torch.Tensor]]:
```

Comment:
mypy might throw a fuss here (since we are now returning a tuple with 3 arrays / tensors. The return should be replaced with

` -> Tuple[Tuple[Union[np.ndarray, torch.Tensor], ...], Union[np.ndarray, torch.Tensor]]`

The ellipses tell mypy the Tuple can have any length

Reply:
Perfect